### PR TITLE
Initial attempt at styled checkboxes/radio buttons.

### PR DESCRIPTION
**Pull request by DFurnes**
# Changes
 - Adds checkbox and radio styles inspired by [WTF, Forms](http://wtfforms.com). These are tested to work in IE9+, Firefox, Chrome, and iOS. IE8 (via conditional class) falls back to platform native styles. Neat!

# Caveat
~~This doesn't seem to work on Android 4.0's stock browser, seemingly due to a bug with adjacent class selectors. Crummy. I'm having trouble finding a non-hacky way of sidestepping this.~~

__Update:__ Added a custom Modernizr test for the weird label behavior. Working nicely now!

# Screenshots

#### Before:
screen shot 2015-01-26 at 3 12 01 pm

#### After:
screen shot 2015-01-26 at 4 17 17 pm

### CODE REVIEW

File: scss/_base/_forms.scss [hunk 1]
```diff
@@ -156,19 +156,97 @@ select {
...
+    background-position: center center;
+    background-repeat: no-repeat;
+    user-select: none;
+  }
+
+  input {
+    position: absolute;
+    opacity: 0;
+    z-index: -1; // Put the input behind the label so it doesn't overlay text
+  }
```
Comment (weerd) [inline scss/_base/_forms.scss]: Maybe use the `@include visually_hidden()` mixin added to Neue to hide this? It doesn't rely on opacity so it also supports IE8 ;)

**Comment by sbsmith86** [comment]
@DFurnes possible to use something other than adjacent classes? like data attributes or roles? Might be a hacky solution but not sure.

**Comment by weerd** [comment]
Does Android support the adjacent sibling selector?

So instead of:
`````
input:checked:active ~ .option__indicator {
  background-color: darken($blue, $tint);
}
`````

use this:
`````
input:checked:active + .option__indicator {
  background-color: darken($blue, $tint);
}
`````

**Comment by DFurnes** [comment]
From doing a bit more research, it looks like Android [doesn't support](http://quirksmode.org/css/selectors/mobile.html) the CSS3 `:checked` selector. I wonder if we can just use a Modernizr test for that?

File: scss/_base/_forms.scss (hunk 1 above)
Comment (DFurnes) [inline scss/_base/_forms.scss]: Another issue is lack of support for `:checked` selector in IE8. :(

**Comment by DFurnes** [comment]
So, there is a [Modernizr test](https://github.com/Modernizr/Modernizr/blob/master/feature-detects/css/checked.js)!

**Comment by weerd** [comment]
Nice! At least you can test for it :)

**Comment by DFurnes** [comment]
So it turns out there's two parts:

1. Lack of support for `:checked` selector. This rules out IE8 and below, and according to MDN Android 2.3 and below. (This is handled by the test linked above, but it won't be available through the builder until 3.0 is released so including it manually for now)
2. Lack of support for triggering an `input` by clicking its associated `label`: this ends up being what breaks Android 4.x browsers. I made a small custom Modernizr test so that we can just disable custom styles on those browsers.

**Comment by weerd** [comment]
Hmmm, I wonder if this will affect the changes I've been doing to the file input in Reportback stuff on Android...?

**Comment by DFurnes** [comment]
Hopefully shouldn't. :grimacing:  All the styles are scoped under `.option` class.

File: scss/_base/_forms.scss [hunk 2]
```diff
@@ -156,19 +156,98 @@ select {
...
-    display: inline;
-    padding-right: $base-spacing;
+  input {
+    position: absolute;
+    top: 2px;
+    left: 0;
   }
 }
 
+.checked.label-click .option {
```
Comment (weerd) [inline scss/_base/_forms.scss]: Might wanna go for that whole `.modernizr-checked` `.modernizr-label-click` class prefixing soon... I fear class collisions! :blue_car: :boom: :car:

**Comment by weerd** [comment]
I meant more, will the reportback file input be affected by this Android bug/lack of support for clicking label to activate input :cry:

File: scss/_base/_forms.scss (hunk 2 above)
Comment (DFurnes) [inline scss/_base/_forms.scss]: For sure! I'm going to do that in another PR so this one doesn't get too big.

**Comment by DFurnes** [comment]
Oh! I hope not... but if so, we have a feature test! :wink:

**Comment by weerd** [comment]
True!

image

**Comment by weerd** [comment]
:+1:
